src/LandIce/problems/LandIce_StokesFOBase.hpp

```diff
@@ -835,133 +1010,16 @@ void StokesFOBase::constructBasalBCEvaluators (PHX::FieldManager<PHAL::AlbanyTra
-    fm0.template registerEvaluator<EvalT> (ev);
-
-    //---- Restrict velocity from cell-based to cell-side-based
-    ev = evalUtils.constructDOFCellToSideEvaluator(dof_names[0],ssName,"Node Vector",cellType,velocity_side);
-    fm0.template registerEvaluator<EvalT> (ev);
-
-    //---- Interpolate velocity on QP on side
-    ev = evalUtils.constructDOFVecInterpolationSideEvaluator(velocity_side, ssName);
-    fm0.template registerEvaluator<EvalT>(ev);
-
-    //---- Interpolate velocity gradient on QP on side
-    ev = evalUtils.constructDOFVecGradInterpolationSideEvaluator(velocity_side, ssName);
-    fm0.template registerEvaluator<EvalT>(ev);
-
-    //---- Compute Quad Points coordinates on the side set
-    ev = evalUtils.constructMapToPhysicalFrameSideEvaluator(cellType, sideCubature[ssName], ssName, enableMemoizer);
-    fm0.template registerEvaluator<EvalT> (ev);
-
-    //---- Interpolate stiffening gradient on QP on side
-    ev = evalUtils.getPSTUtils().constructDOFInterpolationSideEvaluator(stiffening_factor_side, ssName, enableMemoizer);
-    fm0.template registerEvaluator<EvalT>(ev);
-
-    //---- Interpolate stiffening gradient on QP on side
-    ev = evalUtils.getPSTUtils().constructDOFGradInterpolationSideEvaluator(stiffening_factor_side, ssName, enableMemoizer);
-    fm0.template registerEvaluator<EvalT>(ev);
-
-    //---- Interpolate the 3D state on the side (the BasalFrictionCoefficient evaluator needs a side field)
-    ev = evalUtils.getPSTUtils().constructDOFCellToSideEvaluator("stiffening_factor",ssName,"Node Scalar",cellType,stiffening_factor_side);
-    fm0.template registerEvaluator<EvalT> (ev);
-
-    // Intepolate bed_topography
-    ev = evalUtils.getPSTUtils().constructDOFInterpolationSideEvaluator(bed_topography_side, ssName, enableMemoizer);
-    fm0.template registerEvaluator<EvalT> (ev);
-
-    // Intepolate effective_pressure
-    ev = evalUtils.getPSTUtils().constructDOFInterpolationSideEvaluator(effective_pressure_side, ssName, enableMemoizer);
-    fm0.template registerEvaluator<EvalT> (ev);
-
-    //---- Interpolate thickness on QP on side
-    ev = evalUtils.getPSTUtils().constructDOFInterpolationSideEvaluator(ice_thickness_side, ssName, enableMemoizer);
-    fm0.template registerEvaluator<EvalT>(ev);
-
-    //---- Interpolate thickness gradient on QP on side
-    ev = evalUtils.getPSTUtils().constructDOFGradInterpolationSideEvaluator(ice_thickness_side, ssName, enableMemoizer);
-    fm0.template registerEvaluator<EvalT>(ev);
-
-    //---- Interpolate surface height on QP on side
-    ev = evalUtils.getPSTUtils().constructDOFInterpolationSideEvaluator(surface_height_side, ssName);
-    fm0.template registerEvaluator<EvalT>(ev);
-
-    //---- Interpolate basal_friction (if needed) on QP on side
-    basalMemoizer = enableMemoizer ? (!is_dist_param["basal_friction"] ? true : false) : false;
```

Comment:
It might have something to do with this tricky business with the basal boundary conditions. `basalMemoizer` turns on the memoizer for the basal bc evaluators if it's not a distributed parameter.

Reply:
Maybe. Although the performance test does not have any distributed parameters.